### PR TITLE
create multiple pairs issue resolved

### PR DESCRIPTION
**Pull request by abhishek-k-sharma-lab49**
create multiple pairs issue resolved

### CODE REVIEW

File: test/pair-test.ts [hunk 1]
```diff
@@ -89,16 +91,23 @@ describe("All Tests", function () {
...
+      const pairCreateTransaction1 = await factory.createPair(tokenAAddress, tokenBAddress, );
+      const record = await pairCreateTransaction1.wait();
+      const pair1 = record.events[2].args._pairAddress.toString();
+      const pairs = await factory.getPairs(0)
+      expect(pairs[0]).to.be.equals(pair1);
+      const pairCreateTransaction2 = await factory.createPair(tokenAAddress, tokenCAddress);
+      const record2 = await pairCreateTransaction2.wait();
+      const pairs2 = await factory.getPairs(0);
+      const pair2 = record2.events[2].args._pairAddress.toString();
+      expect(pairs2[1]).to.be.equals(pair2);
```
[the remark this answers is not in the document]
Comment (amandeeprehal-lab49) [reply test/pair-test.ts:110]: add the array length check as well.

File: integrationTest/factory.ts [hunk 2]
```diff
@@ -239,28 +263,31 @@ const getTreasureBalance = async (tokens: Array<TokenId>) => {
 }
 
 async function main() {
-    await setupFactory();
-    await testForSinglePair(contractId, tokenA.toSolidityAddress(), tokenB.toSolidityAddress());
+    //await setupFactory();
```
Comment (amandeeprehal-lab49) [inline integrationTest/factory.ts:266]: You need this commented?

File: contracts/Factory.sol [hunk 3]
```diff
@@ -41,8 +40,15 @@ contract Factory is Initializable {
         return address(pair);
     }
 
-    function getPairs() public view returns (IPair[] memory) {
-        return allPairs;
+    function getPairs(uint from) public view returns (address[100] memory) {
```
Comment (amandeeprehal-lab49) [inline contracts/Factory.sol:43]: Add comment on this function to call it multiple times until one reach the end.

File: test/pair-test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhishek-k-sharma-lab49) [reply test/pair-test.ts:110]: Array Size will always be 100.

File: integrationTest/factory.ts [hunk 4]
```diff
@@ -230,6 +232,28 @@ const swapToken = async (contId: string, token: TokenId) => {
   console.log(`Swap status: ${transferTokenRx.status}`);
 };
 
+const getTokenPairAddress = async (contId: string) => {
+  const tokenQty = withPrecision(1);
+  console.log(`Swapping a ${tokenQty} units of token A from the pool.`);
+  // Need to pass different token B address so that only swap of token A is considered.
```
[the remark this answers is not in the document]
Comment (abhishek-k-sharma-lab49) [reply integrationTest/factory.ts:238]: removed

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**